Reconfigure particle flow cfi

hatakeyamak opened the pull request:
#### PR description:

Reconfigure particleFlow_cfi and make clear the parameters used for PFEGammaFilters, PFMuonAlgo, i.e. different components of PFAlgo. Also used this opportunity to remove obsolete config parameters. 

#### PR validation:

Compiles and run a few matrix tests with success (scenarios for 2017, 2018, and phase2)

#### if this PR is a backport please specify the original PR:

This is not a backport PR.

@bendavid @jpata @guitargeek

Commit:
Introduce APV dynamic gain simulation

Use SiStripApvSimulationParameters from EventSetup in SiStripDigitizer

Use SiStrip APV simulation baseline ascii files from SimTracker/SiStripDigitizer/data

SiStripDIgitizerAlgorithm: use enum for tracker partitions

Address PR comments.

Make output collections unique_ptr from start

Use emplace_back instead of push_back.  Reserve size for vectors holding output collections if known

Fix creation of output containers

Reorganise new class members

Fix to avoid potentially negative charge deposits after simulating APV gain.

mmusich committed:
update PVValidation tool as run for 2017 ultra-legacy

mmusich committed:
apply mandatory code-formatting

mmusich committed:
add macro to run-by-run PV Validation trends

cvuosalo committed:
Condensed commit for mf geom builder migration to DD4hep

cvuosalo committed:
Condensed commit for mf geom builder migration to DD4hep

cvuosalo committed:
Remove unneeded CLHEP

cvuosalo committed:
Formatted again

cvuosalo committed:
Formatted again

cvuosalo committed:
Revert unnecessary change to MagGeometryAnalyzer.cc

cvuosalo committed:
Revert unnecessary change to MagGeometryAnalyzer.cc

mmusich committed:
improve plotting macro, get rid of spurious warning messages

mmusich committed:
fix title

cvuosalo committed:
Clarify comment and remove unused data member

cvuosalo committed:
Clarify comment and remove unused data member

hatakeyamak committed:
Clean up the particleFlow config file and make associated changes

hatakeyamak committed:
code-check

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27782/11468

- This PR adds an extra 72KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  RecoParticleFlow/PFProducer/python/particleFlow_cfi.py  modified in PR(s): #27633


Code check has found code style and quality issues which could be resolved by applying following patch(s)
- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27782/11468/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27782/11468/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

slava77 commented on RecoParticleFlow/PFProducer/interface/PFAlgo.h:
```diff
@@ -85,12 +85,7 @@ class PFAlgo {
-                                   double minSignificanceReduction,
-                                   double maxDeltaPhiPt,
-                                   double minDeltaMet);
+				   const edm::ParameterSet& pfHFCleaningParams);
```
there is a preference to reduce dependence on ```ParameterSet``` this is going against that

hatakeyamak replied:
Hmm. What is the reason behind reducing the dependence on ParameterSet?

This PFHFCleaningParameters and also PFBadHcalMitigationParameters are less important, and we don't have to introduce these PSets, but I think it's a good idea to isolate Parameters for PFEGammaFilters and PFMuonAlgo, which I am hoping that EGamma and Muon POGs keep eyes on and manage (factorizing parameter sets to factorize responsibilities).

slava77 replied:
I'm using this https://github.com/cms-sw/cmssw/pull/27010#issuecomment-497804713 as a reference.
Perhaps it does not fully apply in this case: there the tool is delivered via the event setup, while here the tool is created by the ED producer directly.

@Dr15Jones 
is it worth to reduce dependence on the ```ParameterSet``` in the helper algorithms used by ED producers/analyzers?
The factory pattern [not the case here, but somewhat related] even enforces this dependence.

hatakeyamak replied:
Okay. I perfer to leave the PR as it is, but let's see what @Dr15Jones would say.

Dr15Jones replied:
In general, the framework group encourages people to reduce dependencies on the ParameterSet class. It can make it harder to reuse code and the calls to `getParameter` always do parsing so are not super efficient.

hatakeyamak replied:
ok.

hatakeyamak committed:
code-format

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27782/11469

- This PR adds an extra 72KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  RecoParticleFlow/PFProducer/python/particleFlow_cfi.py  modified in PR(s): #27633

cmsbuild commented:
A new Pull Request was created by  @hatakeyamak (Kenichi Hatakeyama) for master.

It involves the following packages:

RecoParticleFlow/PFProducer


@perrotta, @cmsbuild, @slava77 can you please review it and eventually sign? Thanks.
@mmarionncern, @lgray, @seemasharmafnal, @bachtis, @cbernet this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: ccd41d9db7baafeb3f7971832eb1d5c106d364df

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b790fa/2050/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals  AddOn**

* **RelVals**:

When I ran the RelVals I found an error in the following workflows:
140.56 step2
<pre>runTheMatrix-results/140.56_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18/step2_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18.log

</pre>158.0 step2
<pre>runTheMatrix-results/158.0_HydjetQ_B12_5020GeV_2018_ppReco+HydjetQ_B12_5020GeV_2018_ppReco+DIGIHI2018PPRECO+RECOHI2018PPRECO+ALCARECOHI2018PPRECO+HARVESTHI2018PPRECO/step2_HydjetQ_B12_5020GeV_2018_ppReco+HydjetQ_B12_5020GeV_2018_ppReco+DIGIHI2018PPRECO+RECOHI2018PPRECO+ALCARECOHI2018PPRECO+HARVESTHI2018PPRECO.log

</pre>11634.0 step2
<pre>runTheMatrix-results/11634.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2021_GenSimFull+DigiFull_2021+RecoFull_2021+HARVESTFull_2021+ALCAFull_2021/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2021_GenSimFull+DigiFull_2021+RecoFull_2021+HARVESTFull_2021+ALCAFull_2021.log

</pre>12434.0 step2
<pre>runTheMatrix-results/12434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023_GenSimFull+DigiFull_2023+RecoFull_2023+HARVESTFull_2023+ALCAFull_2023/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023_GenSimFull+DigiFull_2023+RecoFull_2023+HARVESTFull_2023+ALCAFull_2023.log

</pre>

* **AddOn**:

I found errors in the following addon tests:

cmsRun /cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc700/cms/cmssw-patch/CMSSW_11_0_X_2019-08-15-2300/src/HLTrigger/Configuration/test/OnLine_HLT_HIon.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Fri Aug 16 11:42:23 2019-date Fri Aug 16 11:41:11 2019 s - exit: 18688
cmsDriver.py RelVal -s HLT:HIon,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_HIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018_pp_on_AA --processName=HLTRECO --filein file:RelVal_Raw_HIon_DATA.root --fileout file:RelVal_Raw_HIon_DATA_HLT_RECO.root : FAILED - time: date Fri Aug 16 11:42:23 2019-date Fri Aug 16 11:41:11 2019 s - exit: 256
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc700/cms/cmssw-patch/CMSSW_11_0_X_2019-08-15-2300/src/HLTrigger/Configuration/test/OnLine_HLT_GRun.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Fri Aug 16 11:48:35 2019-date Fri Aug 16 11:41:16 2019 s - exit: 18688
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run3 --processName=HLTRECO --filein file:RelVal_Raw_GRun_MC.root --fileout file:RelVal_Raw_GRun_MC_HLT_RECO.root : FAILED - time: date Fri Aug 16 11:48:35 2019-date Fri Aug 16 11:41:16 2019 s - exit: 18688
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc700/cms/cmssw-patch/CMSSW_11_0_X_2019-08-15-2300/src/HLTrigger/Configuration/test/OnLine_HLT_HIon.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Fri Aug 16 11:49:50 2019-date Fri Aug 16 11:45:37 2019 s - exit: 18688
cmsDriver.py RelVal -s HLT:HIon,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_HIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018_pp_on_AA --processName=HLTRECO --filein file:RelVal_Raw_HIon_MC.root --fileout file:RelVal_Raw_HIon_MC_HLT_RECO.root : FAILED - time: date Fri Aug 16 11:49:50 2019-date Fri Aug 16 11:45:37 2019 s - exit: 256
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc700/cms/cmssw-patch/CMSSW_11_0_X_2019-08-15-2300/src/HLTrigger/Configuration/test/OnLine_HLT_GRun.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Fri Aug 16 11:51:55 2019-date Fri Aug 16 11:48:42 2019 s - exit: 18688
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018 --processName=HLTRECO --filein file:RelVal_Raw_GRun_DATA.root --fileout file:RelVal_Raw_GRun_DATA_HLT_RECO.root : FAILED - time: date Fri Aug 16 11:51:55 2019-date Fri Aug 16 11:48:42 2019 s - exit: 18688

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

slava77 commented:
@hatakeyamak 
can we introduce proper parameter set validation?
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp
This is a recommended way.
I see that this PR is introducing more "existsAs" cases, which are discouraged and may be OK if there is a good reason for an exception.

hatakeyamak commented:
I will look into it, but will need some time to understand details.

Basically you are suggesting that we introduce fillDescriptions, right?

For parameterSets for PFEGammaFilters, PFMuonAlgo, etc, I am not sure what is the optimal way. Should I try to have fillDescriptions somehow in PFEGammaFilters.cc, PFMuonAlgo.cc, or it had better in PFProducer.cc (the latter would be simpler, but the former may be a better organization conceptually).

slava77 commented:
> I will look into it, but will need some time to understand details.
> Basically you are suggesting that we introduce fillDescriptions, right?

yes, please do.
Thank you.

ianna committed:
GeometricDet migration to DD4hep

ianna committed:
code format

felicepantaleo committed:
Pass rechittools to filter and avoid applying filter to scintillator

slava77 commented:
@hatakeyamak 
please clarify on your current plans for this PR.
Thank you.

hatakeyamak commented:
Plan to make changes you suggested. I slowed down due to grading of exams... Probably I can get it done by sometime next week.

cvuosalo committed:
Addressing first part of Slava's comments. Currently not compiling

cvuosalo committed:
Addressing first part of Slava's comments. Currently not compiling

ianna committed:
Use templates

cvuosalo committed:
Push since IB has expired -- need to rebase. Addressing PR comments, but not compiling yet.

cvuosalo committed:
Compile errors fixed

cvuosalo committed:
Update for latest version of external DD4hep

cvuosalo committed:
Code formatting fixes

slava77 commented:
> Plan to make changes you suggested. I slowed down due to grading of exams... Probably I can get it done by sometime next week.

just for a weekly check, please clarify on the status.
Thank you.
(if it will take longer, perhaps this PR can be closed and then reopened or replaced later when the changes are made).

hatakeyamak commented:
ok. closing for now. but will be working on it later today and tomorrow.

guitargeek committed:
cleanup EgammaHLTAlgos/Producer

guitargeek committed:
indicate which plugin could not be loaded in PluginManager

guitargeek committed:
remove some unused includes in CondCore

guitargeek committed:
code-format CondCore/CondDB/test/testTempWrapper_0.cpp

guitargeek committed:
remove some temporary code

guitargeek committed:
cleanup RecoEgamma/EgammaHLTProducers/plugins/BuildFile.xml

cmsbuild committed:
Merge pull request #27907 from ianna/dd4hep-filtered-view-v6

DD4hep: Filtered View Update

cmsbuild committed:
Merge pull request #27890 from namapane/DTDigi-Phase2Update-11X

Update DTDigi so that it can be accomodate Phase2 TDC units.

Commit:
FixLumisPerJobSplittingFor2018

Commit:
add missing wmsplit of 2018 workflos

Commit:
add missing wmsplit of 2018 workflos

mmusich committed:
implement code-format suggestions

ianna committed:
Code format

smuzaffar committed:
move TreeWriterForEcalCorrection in to a plugin directory

Commit:
Add missing headers for DD4

smuzaffar committed:
move EcalPedestalHistory in to plugins directory to avoid duplicate definitions

mbluj committed:
Monitor deepTau2017 v2p1 instead of deprecated v2

Commit:
Add breaks as discussed in the comments

Commit:
Remove Sequential* from iface and src as discussed

Commit:
Revert L93 and L154 as discussed

felicepantaleo committed:
Adding possibility to perform Out-In search \
in the Trackster creation by Depth-first search

felicepantaleo committed:
undo cut changes in TICL_iterations

cmsbuild committed:
Merge pull request #27582 from PFCal-dev/hgc_eolt_11_0_X

Improvements to the HGC digitizer and implementation of a end-of-life scenario for the HGC silicon sector

cmsbuild committed:
Merge pull request #27522 from guitargeek/GsfEleEBEECut

New GsfEleEBEECut class to generalize some ElectronIdentification cuts

cmsbuild committed:
Merge pull request #27821 from perrotta/removeDeadAssignments

Remove dead assignments in a few sparse reco packages

cmsbuild committed:
Merge pull request #27900 from cardinia/from-CMSSW_10_5_0_pre2

Updated tool for Tracker Alignment: DrmsNR trends

cmsbuild committed:
Merge pull request #27914 from mrodozov/add-headers-dd4-pkg

Add missing headers for DD4

Commit:
Update break to return SiStripModuleGeometry::UNKNOWNGEOMETRY as discussed in comments

guitargeek committed:
address comments

gouskos committed:
updates

ianna committed:
Move the files to an approved structure

cmsbuild committed:
Merge pull request #27425 from sbein/CharginosInMuonSystem

updated FastSim treatment of long-lived charginos in the muon system

cmsbuild committed:
Merge pull request #27835 from perrotta/removeLogicErrorAndDeadStoresInLocalMuons

Try to remove a few LogicErrors and DeadStores in DT/CSC LocalMuon reco

cmsbuild committed:
Merge pull request #27878 from cms-tau-pog/CMSSW_11_0_X_tau-pog_deepTauVetoPCA

Disabled PCA inputs for deepTau v2

cmsbuild committed:
Merge pull request #27913 from smuzaffar/fix-plugin-TreeWriterForEcalCorrection

move TreeWriterForEcalCorrection in to a plugin directory

cmsbuild committed:
Merge pull request #27910 from srimanob/11_FixLumisPerJobSplittingFor2018

FixLumisPerJobSplittingFor2018

cmsbuild committed:
Merge pull request #27746 from srimanob/11X_UpdateAutoSkimForEGammaDatasets

Update autoskim for egamma datasets

cmsbuild committed:
Merge pull request #27828 from Dr15Jones/restructureL1TriggerGlobalMuonTrigger

Removed use of L1Trigger/GlobalMuonTrigger from test

cmsbuild committed:
Merge pull request #27915 from smuzaffar/plugin-fix-EcalPedestalHistory

move EcalPedestalHistory in to plugins directory to avoid duplicate definitions

cmsbuild committed:
Merge pull request #27843 from mrodozov/fixgcc9warns-CommonTools

Fix gcc9 warnings in CommonTools

cmsbuild committed:
Merge pull request #27912 from mmusich/fixPremixingForStuckTBM

Fix simulation of the pixel bad components on the FED channel basis for PreMixing

cmsbuild committed:
Merge pull request #27797 from mrodozov/fix-gcc9-warnings-HetCore-CudaSrvcs

Fix warnings for gcc9 IB for HeterogeneousCore/CUDAServices

wddgit committed:
Define move assignment in SoATuple.h

Silences clang compiler warning.

kpedro88 committed:
fix handling of intermediate lumi values in HCAL aging

cvuosalo committed:
Fix second round of PR comments

christopheralanwest committed:
Updates to beamspot, tracker alignment, muon alignment, ECAL calibrations and a fix for pixel conditions for a single fill.

christopheralanwest committed:
Updated PPS conditions for consistency with 2018 UL

christopheralanwest committed:
Updated 2018 MC tracker alignment scenario and APEs

christopheralanwest committed:
Change to EcalADCToGeVConstantRcd tag populated by O2O

cvuosalo committed:
Part 2 of fixing second round of comments

aspiezia committed:
TAU miniaod validation/dqm

aspiezia committed:
style corrections

guitargeek committed:
add context with plugin name to PluginLibraryLoadError

ianna committed:
Template all

ianna committed:
Cleanup

ianna committed:
Apply code format

gouskos committed:
address Slava's comments

gouskos committed:
fix style issues

christopheralanwest committed:
Make PPSDiamondTimingCalibration tag consistent with tag in offline GT.

Dr15Jones committed:
Explicitly removed methods implicitly ignored by compiler in FWCore/Framework

Although we had asked the compiler explicitly to create default versions
of functions, it was implicitly deleting them. Now we explicitly
request them to be deleted to match what the compiler was doing.

This fixes new clang warnings.

fabiocos committed:
Add an LHE-based mc relval with multiple lumi sections and concurrent lumi settings

ianna committed:
Pass a reference

fabiocos committed:
Update Reconstruction_Data for move of jetCorrectorsForReco to Task

mundim committed:
Applying code style changes from PR 27673 comments

mundim committed:
Removing a typo in a LogDebug message

cmsbuild committed:
Merge pull request #27818 from mrodozov/fix-gcc9-warnings-L1Trigger/L1TCalorimeter

Fix gcc 9 warnings in L1Trigger/L1TCalorimeter

cmsbuild committed:
Merge pull request #27785 from hatakeyamak/Fix_HGCalSuperclusterVetoes

Properly veto HGCal superclusters from entering regular PFBlocks - phase2 scenarios

mundim committed:
including a forgotten type cast in a for loop when removing a hardcoded value

cmsbuild committed:
Merge pull request #27923 from christopheralanwest/alca-final-2018-gt

Data and MC global tags for 2018 UL

cmsbuild committed:
Merge pull request #27931 from fabiocos/fix-140d54

Update Reconstruction_Data for move of jetCorrectorsForReco to Task

cmsbuild committed:
Merge pull request #27920 from ianna/dd4hep-tracker-algos-v7

DD4hep: Move Files to an Approved Directory Structure

cmsbuild committed:
Merge pull request #27833 from clelange/JetValidationPlots

HGCal and jets validation plots harvesting

cmsbuild committed:
Merge pull request #27892 from mmusich/SiPixelLorentzAngleMaps

SiPixel Payload Inspector improvements

cvuosalo committed:
Namespace adjustments

ianna committed:
Address Vincenzo's comments

ianna committed:
Merge branch 'master' into dd4hep-geom-geometric-det-v3

apsallid committed:
adding HGCal support

apsallid committed:
creating python script for HGCal analyser

apsallid committed:
Adding HGCal support to the producer

apsallid committed:
The dEdx weights calculation script

pieterdavid committed:
Fix a bug (typo) in a constructor of PhysicsTools::Calibration::HistogramF3D

apsallid committed:
applying code-checks

fabiocos committed:
Update structure according to Kevin's suggestion, move concurrent lumi settings into a global variable

fabiocos committed:
Add a data based relval test wf with lumi boundary crossing

Commit:
Make changes to conform G4/GV Comparison

sbein committed:
Added safety statements for tracker hits that are not associated to a cluster to avoid seg faults, enabled HSCP treatment to be applied to R-hadrons

cvuosalo committed:
Code formatting

pieterdavid committed:
Add SiStripApvSimulationParameters and associated record

including an ESSource, and EDAnalyzer to add to the database

Commit:
Correct the flag

Commit:
Conevrt one DDHGCalNoTaperEndcap algorithm to DD4Hep format

sbein committed:
ran code-checks to overwrite my files

pieterdavid committed:
code-format fixes

Commit:
Code check

Commit:
Remove reference to G4DataQuestionaire

Commit:
Codecheck

hmiaozh committed:
one line fix for mahi algorithm

Sam-Harper committed:
fixing returned value for heep trk isol

cmsbuild committed:
Merge pull request #27844 from mrodozov/fixgcc9warns-DataFormats

Fix gcc 9 warnings in DataFormats

cmsbuild committed:
Merge pull request #27921 from wddgit/soaMoveAssign

Define move assignment in SoATuple.h

cmsbuild committed:
Merge pull request #27854 from mrodozov/fixgcc9warn-L1TriggerConfig

Fix gcc9 warning in L1TriggerConfig

cmsbuild committed:
Merge pull request #27901 from jpata/pfalgo_elementloop_factorize

simplify elementLoop flow, add comments

cmsbuild committed:
Merge pull request #26592 from wddgit/concurrentIOVs1May2019R

Implement core support for concurrent IOVs

cmsbuild committed:
Merge pull request #27927 from Dr15Jones/clangWarningsFWCoreFramework

Explicitly removed methods implicitly ignored by compiler in FWCore/Framework

ianna committed:
Legacy shape support

ianna committed:
Code format

fabiocos committed:
add a workflow for lumi boundayr crossing within the same run

wddgit committed:
Add DataProxy::initializeForNewIOV

This notifies DataProxy's when a new IOV actually
starts. It will be important and needed for CondDBESSource
when it is upgraded to support concurrent IOVs. That
is the only known use case. For everything else
currently existing, this function will do nothing.
As a performance optimization only, there is a
restriction that this only gets called for records
with finders.

wddgit committed:
add virtual DataProxyProvider::initConcurrentIOVs

This new function intended for use by CondDBESSource when
it is upgraded to support concurrent IOVs. Nothing currently
overrides this and by default it does nothing. Early in
the job while things are being initialized it lets
DataProxyProviders know how many concurrent IOVs there
will be for each record. CondDBESSource needs to know
this to be able to create its DataProxy's.

wddgit committed:
setIntervalFor run in EventSetup mutex lock

wddgit committed:
Be more careful about resetting lumi status

wddgit committed:
Check number of configured concurrent IOVs

Don't let it exceed the number of streams or
the limit on the number of concurrent lumis.
Reset it to a smaller value if it is too big
because that would just be useless waste (it
wouldn't fail, just unecessary use of CPU
and memory).

wddgit committed:
Add new unit test for concurrent IOV code

wddgit committed:
Fix merging mistake in previous commit

cmsbuild committed:
Merge pull request #27841 from davidlange6/190823_boostsharedptr

move away from boost shared_ptrs

cvuosalo committed:
Fix formatting of debug messages

hmiaozh committed:
code-format

ianna committed:
remove duplicates

fabiocos committed:
Fix headers for check-headers errors reported in PR 27841

Dr15Jones committed:
Ignore sanitize options when generating CondFormat serialzations

The sanitize compiler options are not useful when all we need is
the compiler to do C++ parsing, not code generation. Ignoring the
option avoids problems when clang does not support the same options
as gcc.

cmsbuild committed:
Merge pull request #27951 from fabiocos/fc-fix-headers2-20190907

Fix headers for check-headers errors reported in PR 27841

cmsbuild committed:
Merge pull request #27943 from bsunanda/Run3-sim45

Run3-sim45 Remove reference to G4DataQuestionaire

cmsbuild committed:
Merge pull request #27944 from Sam-Harper/HEEPVIDTrkIsoValueFix_1100pre7

Fixing VID return value for HEEP Trk Iso: 11_0_X

cmsbuild committed:
Merge pull request #27939 from sbein/from-CMSSW_11_0_X_2019-09-04-2300

Minor technical fix for FastSim de/dx LLP

cmsbuild committed:
Merge pull request #27850 from mrodozov/fixgcc9warn-HeavyFlavorAnalysis/RecoDecay

Fix gcc9 warning in HeavyFlavorAnalysis/RecoDecay

cmsbuild committed:
Merge pull request #27852 from mrodozov/fixgcc9warn-MuonAnalysis/MuonAssociators

Fix gcc9 warning in MuonAnalysis/MuonAssociators

cmsbuild committed:
Merge pull request #27827 from mrodozov/fixgcc9warns-L1Trigger/GlobalCaloTrigger

Fix gcc9 warnings L1Trigger/GlobalCaloTrigger

cmsbuild committed:
Merge pull request #27925 from aspiezia/DQMTauMiniaodValidation_CMSSW110X

TAU miniaod validation/dqm

cmsbuild committed:
Merge pull request #27673 from CTPPS/FullSim_SimG4CMS-PPS

SimG4CMS/PPS module for the PPS full simulation

cmsbuild committed:
Merge pull request #27855 from mrodozov/fixgcc9warnings-PhysicsTools

Fix gcc9 warning in PhysicsTools

fabiocos committed:
Move SimG4CMS/PPS dependency to geant4core only

apsallid committed:
removing cout and cerr

ianna committed:
Strip off residual namespace

EmyrClement committed:
code-format patch

cmsbuild committed:
Merge pull request #25239 from gartung/SimDataFormats-GeneratorProducts-strictcxx17

Convert auto_ptr to unique_ptr in SimDataFormats/GeneratorProducts.

cmsbuild committed:
Merge pull request #27953 from fabiocos/fc-fixbuildfile

Move SimG4CMS/PPS dependency to geant4core only

cmsbuild committed:
Merge pull request #27952 from Dr15Jones/fixSerialization

Ignore sanitize options when generating CondFormat serialzations

cmsbuild committed:
Merge pull request #27922 from kpedro88/FixHcalAging

fix handling of intermediate lumi values in HCAL aging

cmsbuild committed:
Merge pull request #27908 from guitargeek/PluginLoadError

More verbose error if plugin could not be loaded due to missing shared library

cmsbuild committed:
Merge pull request #27928 from fabiocos/fc-relval

Add an LHE-based mc relval with multiple lumi sections and concurrent lumi settings

cmsbuild committed:
Merge pull request #27909 from guitargeek/EgammaHLT

Cleanup of EgammaHLTAlgos/Producers

kpedro88 committed:
fix SimProducers for MT

gouskos committed:
resolve Kevin's comment + bug fix on ieta loop

gouskos committed:
code-formats

Commit:
Remove fallthrough and add break as suggested in comments

Dr15Jones committed:
Added unit test for HLTrigReport

Dr15Jones committed:
Support concurrent LuminosityBlocks in HLTrigReport

Switched to using a LuminosityBlockCache to allow HLTrigReport
to not be sensitive to the ordering of completion of a LuminosityBlock.
This allows concurrent LuminosityBlocks to be processed properly.

Support for concurrent Runs would require further work.

pieterdavid committed:
SiStripApvSimulationParameters: add members for TID and TEC partitions

wddgit committed:
Add comment only

wddgit committed:
Rename initialize functions, add LogInfo message

Dr15Jones committed:
Explicitly define move related ReferenceCounted methods

The clang compiler warned that the '= default' version were not
begin generated by the compiler. Fixed via explicit versions.

Commit:
Add step length information for passive hits

Commit:
Code check

cmsbuild committed:
Merge pull request #27847 from mrodozov/fixgcc9warns-FastSimDataFormats

Fix gcc9 warnings in FastSimDataFormats/L1GlobalMuonTrigger

cmsbuild committed:
Merge pull request #27938 from bsunanda/Run3-sim44

Run3-sim44 Make changes to conform G4/GV Comparison

cmsbuild committed:
Merge pull request #27950 from hmiaozh/MahiOneLineFix

HBHE: bugfix for MAHI on issue of missing SOI+4 subtraction

cmsbuild committed:
Merge pull request #27954 from ianna/dd4hep-filtered-view-v7

DD4hep: Strip off Residual Namespace while Calculating the History

cmsbuild committed:
Merge pull request #27946 from ianna/dd4hep-legacy-shape-support

DD4hep: Legacy DD Shape ID Support and Mapping

cmsbuild committed:
Merge pull request #27940 from bsunanda/Run3-hgx218

Run3-hgx218 Conevrt one DDHGCalNoTaperEndcap algorithm to DD4Hep format

cmsbuild committed:
Merge pull request #27792 from depasse/SRP

ECAL : SRP and TPG tools

cmsbuild committed:
Merge pull request #27926 from ianna/dd4hep-geom-geometric-det-v3

DD4hep: Tracker Builder, Numbering and GeometricDet pre-migration

smuzaffar committed:
Run3-sim45 Remove reference to G4DataQuestionaire

Commit:
Put the macro as well

cvuosalo committed:
Add conversion of second, new DDSolidShape related to DD4hep

gouskos committed:
fix

davidlange6 committed:
remove histogrammar (obsolete package)

gouskos committed:
fix

toropin committed:
Comments to parameters definition for HFStripFilter

cmsbuild committed:
Merge pull request #27937 from fabiocos/fc-relval-data

Add a data based relval test wf with lumi boundary crossing

toropin committed:
applied 'scram build code-format'

Martin-Grunewald committed:
Migration to 11_0_0_pre7 template

toropin committed:
Added comments to descriptions using setComment method

toropin committed:
Rearranged comments for descriptions

toropin committed:
Rearranged comments for descriptions

mmusich committed:
document MultiRunAndPlotPVValidation

Commit:
Convert DDHGCalTBModule algorithm for DD4Hep

Commit:
Code Check

pieterdavid committed:
SiStripApvSimulationParameters: rename ring->wheel, z->r

pieterdavid committed:
Update ESProducer for TID and TEC

cmsbuild committed:
Merge pull request #27975 from cms-tsg-storm/HLTmigrationTo1100pre7

Migration of HLT menus to CMSSW_11_0_0_pre7 template

cmsbuild committed:
Merge pull request #27965 from cms-sw/smuzaffar-patch-1

Run3-sim45 Remove reference to G4DataQuestionaire

cmsbuild committed:
Merge pull request #27948 from wddgit/coreSupportForConcurrentDBESSource

Core support for concurrent IOVs in CondDBESSource

cmsbuild committed:
Merge pull request #27963 from Dr15Jones/clangFixBasicReferenceCounted

Explicitly define move related ReferenceCounted methods

mmusich committed:
improved documentation, add dates in the plotting trend macro, general clean-up

EmyrClement committed:
Add functionality to only simulate APV for fraction of events, and store bool with decision.

EmyrClement committed:
code-format patch

smorovic committed:
reintroduce missing lseek() for header-less files

mmusich committed:
several miscelleanous improvements in the plotting macro

mmusich committed:
fix the logic of the parallel jobs splitting

mmusich committed:
code-format

cmsbuild committed:
Merge pull request #27969 from davidlange6/dl190911

remove histogrammar (obsolete package)

cmsbuild committed:
Merge pull request #27959 from kpedro88/FixSimMT

fix SimProducers for MT

cmsbuild committed:
Merge pull request #27964 from bsunanda/Run3-sim46

Run3-sim46 Add step length information for passive hits

cmsbuild committed:
Merge pull request #27856 from mrodozov/figxgcc9warns-Reco

Fix gcc9 warnings in Reco

cmsbuild committed:
Merge pull request #27971 from toropin/comments-for-HFStripFilter

Comments to parameters definition for HFStripFilter

cmsbuild committed:
Merge pull request #27819 from gouskos/hgcal_paramOutIn

HGCAL: 1st version of hadronic reconstruction

cmsbuild committed:
Merge pull request #27888 from guitargeek/MVATrainer

Removing the PhysicsTools/MVATrainer package

cmsbuild committed:
Merge pull request #27961 from Dr15Jones/concurrentLumiHLTrigReport

Support concurrent LuminosityBlocks in HLTrigReport

cmsbuild committed:
Merge pull request #27979 from bsunanda/Run3-hgx220

Run3-hgx220 Convert DDHGCalTBModule algorithm for DD4Hep

Dr15Jones committed:
Removed unimplemented move constructor from StorageAccount

Although the move constructor was marked '=default' the compiler
was unable to generate it. Given all member data of the class
as atomics of builtin types, a move constructor would be no
different from a copy constructor anyway.

Dr15Jones committed:
Explicitly declare copy assignment for bqueue

The clang compiler was warning about classes which had bqueue
objects as members were unable to implement copy assignment
operators because bqueue had not explicitly declared one.

Dr15Jones committed:
Removed implicitly dropped move assignment from RectangularEtaPhiTrackingRegion

The compiler was implictly not implementing the move assignment
operator even though it was marked as '=default'. The reason is
there is a const member data which prohibits any assignments.

cvuosalo committed:
Move newln into magneticfield namespace

cvuosalo committed:
One blank line removed by code-format

cmsbuild committed:
Merge pull request #27935 from apsallid/dedxwei

[HGCal] dEdx weights and plots

cmsbuild committed:
Merge pull request #27748 from abdoulline/Adding_RecHits_spectra_wCleaning

HCAL DQMOffline: adding rechits spectra with enabled cleaning

cmsbuild committed:
Merge pull request #27984 from mmusich/update_aligment_PV_Validation_110X_v3

Tracker Alignment: update PV Validation tool as used in Ultra-Legacy

Commit:
modify edm::propagate_const to enforce const-correctness more strictly, based on the gcc and llvm implementations of the proposed standard

Commit:
formatting

cmsbuild committed:
Merge pull request #27991 from Dr15Jones/clangFixUtilitiesStorageFactory

Removed unimplemented move constructor from StorageAccount

cmsbuild committed:
Merge pull request #27992 from Dr15Jones/clangFix_bqueue

Explicitly declare copy assignment for bqueue

cmsbuild committed:
Merge pull request #27993 from Dr15Jones/fixClangRectangularEtaPhiTrackingRegion

Removed implicitly dropped move assignment from RectangularEtaPhiTrackingRegion

cmsbuild committed:
Merge pull request #27994 from dan131riley/stricter-propagate_const

stronger edm::propagate_const const-correctness

cmsbuild committed:
Merge pull request #27823 from EmyrClement/APVSimulation_PR_master

Simulation of APV dynamic gain

cmsbuild committed:
Merge pull request #27941 from pieterdavid/addSiStriApvSimulationParameters

Add SiStripApvSimulationParameters database object and record

cmsbuild committed:
Merge pull request #27982 from fwyzard/fix_EvFDaqDirector

reintroduce missing lseek() for header-less files

cmsbuild committed:
Merge pull request #27886 from cvuosalo/mfgb-to-dd4hep-for-pr

Magnetic field geometry builder DD4hep migration (part 1)

makortel committed:
Fix ExpandVisitor.result() to work also with an empty result

If a cms.Sequence holds only tasks, the list of sequenceable elements
is empty, and self.l[0] leads to an IndexError.

cmsbuild committed:
Merge pull request #28006 from makortel/fixExpandVisitorEmpty

Fix ExpandVisitor.result() to work also with an empty result

hatakeyamak committed:
introduce fillDescriptions

hatakeyamak committed:
solve conflicts

hatakeyamak committed:
solve conflicts

hatakeyamak committed:
cleanup. reduce parametersets usage somewhat.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27782/12124

- This PR adds an extra 208KB to repository

- Found files with invalid states:
  - MagneticField/GeomBuilder/plugins/dd4hep/MagGeoBuilder.cc:
    - Added: 957d80d086fb0b5622f59510ed3b08429459e353, 6aa8543b01656d69cf8fcf455de29de9f0bf3ac1
    - Modified: e6d1e35a04f7d8713ab890e9914a2e2433d99672, 9cc0070ba23adf0a8c23d43a49f0651e5fb790db
    - Deleted: 786571247f9faa4601aad17d53170b4186645eef, b65032059b281e48f101ebee19e1b21339018998
  - MagneticField/GeomBuilder/plugins/dd4hep/volumeHandle.h:
    - Added: 957d80d086fb0b5622f59510ed3b08429459e353, 6aa8543b01656d69cf8fcf455de29de9f0bf3ac1
    - Deleted: 786571247f9faa4601aad17d53170b4186645eef, b65032059b281e48f101ebee19e1b21339018998
  - MagneticField/GeomBuilder/plugins/dd4hep/VolBasedMagFieldESProducerNewDD.cc:
    - Added: 957d80d086fb0b5622f59510ed3b08429459e353, 6aa8543b01656d69cf8fcf455de29de9f0bf3ac1
    - Modified: eaaed4b685055dba1bc7a64783da7c3aefb24b15, 78d71aff0afc8ebee1e4444acebe69a0a93c1336
    - Deleted: 786571247f9faa4601aad17d53170b4186645eef, b65032059b281e48f101ebee19e1b21339018998
  - MagneticField/GeomBuilder/plugins/dd4hep/MagGeoBuilder.h:
    - Added: 957d80d086fb0b5622f59510ed3b08429459e353, 6aa8543b01656d69cf8fcf455de29de9f0bf3ac1
    - Deleted: 786571247f9faa4601aad17d53170b4186645eef, b65032059b281e48f101ebee19e1b21339018998
  - RecoEgamma/EgammaHLTProducers/interface/Poison.h:
    - Added: 327b1acadd4e3b486759c8dc1edbe858978826c3
    - Deleted: 2bae36bed6b7f8573a946cd95e16868e79b4193f
  - RecoEgamma/EgammaHLTProducers/src/Poison.cc:
    - Added: 327b1acadd4e3b486759c8dc1edbe858978826c3
    - Deleted: 2bae36bed6b7f8573a946cd95e16868e79b4193f
  - MagneticField/GeomBuilder/plugins/dd4hep/volumeHandle.cc:
    - Added: 957d80d086fb0b5622f59510ed3b08429459e353, 6aa8543b01656d69cf8fcf455de29de9f0bf3ac1
    - Modified: eaaed4b685055dba1bc7a64783da7c3aefb24b15, 78d71aff0afc8ebee1e4444acebe69a0a93c1336
    - Deleted: 786571247f9faa4601aad17d53170b4186645eef, b65032059b281e48f101ebee19e1b21339018998

cmsbuild commented:
Pull request #27782 was updated. @SiewYan, @andrius-k, @lveldere, @chayanit, @emeschi, @sbein, @schneiml, @ianna, @kpedro88, @Martin-Grunewald, @rekovic, @fioriNTU, @tlampen, @alberto-sanchez, @pohsun, @santocch, @peruzzim, @perrotta, @civanch, @zhenhu, @cmsbuild, @agrohsje, @fwyzard, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @ssekmen, @mdhildreth, @jfernan2, @tocheng, @slava77, @ggovi, @qliphy, @fabiocos, @mkirsano, @benkrikler, @efeyazgan, @kmaeshima, @christopheralanwest, @pgunnell, @franzoni, @fgolf, @mommsen can you please check and sign again.
